### PR TITLE
[NSMutableArray] replace objects in range docs/tests/impl

### PR DESCRIPTION
**Pull request by cullenbmacdonald**
I think I found a bug while writing these tests. Not sure what the process is for filing it, but I was getting a a complier error "Int does not conform to expected element type AnyObject"  when doing `NSMutableArray(array: [1, 2, 3])` which seems...wrong.

Tested on latests official xcode build from app store and it definitely works.

Will be happy to try and resolve that issue before this gets merged as the tests look a little verbose as a result of the bug.

### CODE REVIEW

File: Foundation.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1041,36 +1043,37 @@
 		EA66F65A1BF1976100136161 /* Tests */ = {
 			isa = PBXGroup;
 			children = (
```
Comment (cullenbmacdonald) [inline Foundation.xcodeproj/project.pbxproj]: simply sorted the order here.

File: Foundation/NSArray.swift [hunk 2]
```diff
@@ -658,7 +658,19 @@ public class NSMutableArray : NSArray {
             }
         }
     }
-    public func replaceObjectsInRange(range: NSRange, withObjectsFromArray otherArray: [AnyObject], range otherRange: NSRange) { NSUnimplemented() }
+    
+    /// Replaces the objects in the receiving array specified by one given range with the objects in another array specified by another range.
```
Comment (cullenbmacdonald) [inline Foundation/NSArray.swift]: these docstrings are copy pasted from official docs.

File: TestFoundation/TestNSMutableArray.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+class TestNSMutableArray : XCTestCase {
+    
+    var allTests : [(String, () -> ())] {
+        return [
+            ("test_replaceObjectsInRangeWithObjectsFromArray", test_replaceObjectsInRangeWithObjectsFromArray),
+        ]
+    }
+    
+    func test_replaceObjectsInRangeWithObjectsFromArray() {
+        let existing = [NSNumber(int: 1), NSNumber(int: 2), NSNumber(int: 3), NSNumber(int: 4), NSNumber(int: 5), NSNumber(int: 6)]
```
Comment (cullenbmacdonald) [inline TestFoundation/TestNSMutableArray.swift]: this is what is more verbose as a result of the bug i found.

**Comment by parkera** [comment]
What you're seeing is the lack of implicit bridging from Int to NSNumber. `AnyObject` requires that the type is a class type. Int is not a class type unless it's bridged to an NSNumber.

Unfortunately this is expected behavior until we can get implicit bridging figured out when the ObjC runtime is not present.

**Comment by cullenbmacdonald** [comment]
got it. that makes a lot of sense. Do I need to change anything about this pull then?

File: Foundation/NSArray.swift [hunk 4]
```diff
@@ -658,7 +658,19 @@ public class NSMutableArray : NSArray {
...
+    /// - parameter range: The range of objects to be replaced in (or removed from) the receiving array.
+    /// - parameter otherArray: The array of objects from which to select replacements for the objects in `range`.
+    /// - parameter otherRnage: The range of objects be selected from `otherArray` as replacements for the objects in `range`.
```
Comment (parkera) [inline Foundation/NSArray.swift]: typo here in `otherRnage`

**Comment by parkera** [comment]
Other than the typo, looks good to me.

**Comment by cullenbmacdonald** [comment]
fixed

**Comment by joshuatbrown** [comment]
It looks like there are at least three implementations of this method now:

1. PR #125: https://github.com/apple/swift-corelibs-foundation/pull/125
2. PR #136: https://github.com/apple/swift-corelibs-foundation/pull/136 (this one)
3. PR #139: https://github.com/apple/swift-corelibs-foundation/pull/139

**Comment by thii** [comment]
From the [Contribution Guidelines](https://github.com/apple/swift-corelibs-foundation/blob/master/CONTRIBUTING.md):
> Before embarking on a large amount of work to implement missing functionality, please double-check with the community on the swift-corelibs-dev mailing list. Someone may already be working in this area, and we want to avoid duplication of work.

**Comment by cullenbmacdonald** [comment]
My mistake. Dunno how I missed #125! ill be more careful next time

**Pull request closed without merging**